src/main/java/StringAddCalculator.java

```diff
@@ -0,0 +1,65 @@
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * 문자열로 된 숫자들을 특정 구분자로 분리하여 합산하는 기능을 제공합니다.
+ * 기본 구분자는 컴마(,)와 콜론(:)이며, 커스텀 구분자도 지원합니다.
+ * 음수 입력값에 대해서는 예외를 발생시킵니다.
+ */
+public class StringAddCalculator {
+    private static final String CUSTOM_DELIMITERS_EXTRACT_REGEX = "//(.)\n(.*)";
+    private static final String DEFAULT_DELIMITERS = ",|:";
+
+    public static int splitAndSum(String input) {
+        if (isNullOrEmpty(input)) {
+            return 0;
+        }
+
+        return calcWithDelimiter(input);
+    }
+
+    private static boolean isNullOrEmpty(String input) {
+        return input == null || input.isEmpty();
+    }
+
+    private static int calcWithDefaultDelimiters(String input) {
+        String[] tokens = input.split(DEFAULT_DELIMITERS);
+        return sum(tokens);
+    }
+
+    private static int calcWithCustomDelimiters(String input) {
+        Matcher m = Pattern.compile(CUSTOM_DELIMITERS_EXTRACT_REGEX).matcher(input);
+        if (m.find()) {
+            String customDelimiter = m.group(1);
+            String[] tokens = m.group(2).split(customDelimiter);
+            return sum(tokens);
+        }
+        throw new IllegalArgumentException("입력 형식이 올바르지 않습니다.");
+    }
+
+    private static int calcWithDelimiter(String input) {
+        if (hasCustomDelimiter(input)) {
+            return calcWithCustomDelimiters(input);
+        }
+        return calcWithDefaultDelimiters(input);
+    }
+
+    private static boolean hasCustomDelimiter(String input) {
+        return input.startsWith("//");
+    }
+
+    private static int sum(String[] tokens) {
+        return Arrays.stream(tokens)
+                .mapToInt(Integer::parseInt)
+                .map(StringAddCalculator::validateInt)
+                .sum();
+    }
+
+    private static int validateInt(int number) {
+        if (number < 0) {
+            throw new IllegalArgumentException("음수 입력값은 허용되지 않습니다.");
+        }
+        return number;
+    }
```

Comment:
```suggestion
    private static int toPositive(String value) {
        int number = Integer.parseInt(value);
        if (number < 0) {
            throw new RuntimeException("음수를 입력할 수 없습니다.");
        }
        return number;
    }
```
위와 같이 구현하는 것은 어떻게 생각하나요?

Reply:
- 음수가 아닌 지 검증하는 것
 - String 을 int 로 변환하는 것

두 가지 내용이 하나의 메서드로 들어간다고 생각해서 초반에 말씀하셨던 대로 구현했다가 검증하는 로직만 따로 뺐던 것이긴 한데,
지금의 구조는 확실히 숫자 sum 과 String -> int 변환을 같은 메서드에서 하고 있는 것이라 말씀하신대로 toPositive가 확실히 더 나을 것 같네요..!